test/object.cpp

```diff
@@ -1530,21 +1498,20 @@ class object_test
     void
     testAllocation()
     {
-        counting_resource res;
+        auto& default_resource = *detail::default_resource::get();
+        checking_resource res(default_resource);
         {
             storage_ptr sp(&res);
             object o(sp);
             o.reserve(1);
         }
-        BOOST_TEST(res.allocated == res.deallocated);
 
-        res = {};
+        res = checking_resource(default_resource);
```

Comment:
Each of these scopes should look something like this:
```
    {
        auto sp = make_shared_resource<checking_resource>();
        
        // this should also work
        //auto sp = make_shared_resource<checking_resource>( storage_ptr() );

        object o( sp );
        o.reserve(1);
}
```

Reply:
Alternatively, if you want a non-counted resource:
```
Each of these scopes should look something like this:
```
    {
        checking_resource mr;
        
        // this should also work
        //checking_resource mr( storage_ptr() );

        object o( &mr );
        o.reserve(1);
    }
```